timely/src/dataflow/operators/core/rc.rs

```diff
@@ -27,13 +28,13 @@ pub trait SharedStream<S: Scope, C: Container> {
 impl<S: Scope, C: Container> SharedStream<S, C> for StreamCore<S, C> {
     fn shared(&self) -> StreamCore<S, Rc<C>> {
         let mut container = Default::default();
-        self.unary(Pipeline, "Shared", move |_, _| {
+        self.unary::<BufferingContainerBuilder<_>, _, _, _>(Pipeline, "Shared", move |_, _| {
```

Comment:
We discussed, and it seems possible that this could be a `CapacityContainerBuilder` and we largely use the old code, using `give_container` and prompting a pushthrough.